xtask/src/main.rs.liquid

```diff
@@ -98,6 +100,16 @@ fn check_commits(ctx: &mut Context) {
     }
 }
 
+fn check_doc(ctx: &mut Context) {
+    ctx.sh.set_var("RUSTDOCFLAGS", "-D warnings");
+
+    action!(
+        ctx,
+        "Checking that the documentation builds without warnings",
+        "cargo doc"
+    );
```

Comment:
**question (blocking):** Actually, several questions:

1. Does this work for all project layouts? Especially I’m not sure it works for the Embedded templates.
2. Does this builds the docs for all crates in the repo? In the case there is no workspace, we should handle this.
3. Does this also check the documentation of private items?